types/basic.go

```diff
@@ -5,22 +5,26 @@ import (
 	"fmt"
 	"reflect"
 	"sync"
-	"time"
 
-	"github.com/karlseguin/ccache"
+	"github.com/dgraph-io/ristretto"
 )
 
 // BasicTypeCacheSize for HashTreeRoot.
 const BasicTypeCacheSize = 100000
 
 type basicSSZ struct {
-	hashCache *ccache.Cache
+	hashCache *ristretto.Cache
 	lock      sync.Mutex
 }
 
 func newBasicSSZ() *basicSSZ {
+	cache, _ := ristretto.NewCache(&ristretto.Config{
+		NumCounters: BasicTypeCacheSize, // number of keys to track frequency of (1M).
+		MaxCost:     1 << 23,            // maximum cost of cache (10MB).
```

Comment:
Same, can these all share the same cache?

Reply:
Not too sure if that can be done, @rauljordan can you confirm that it wouldn't lead to any invalid cache values ?